### PR TITLE
[1822] Better logging when phase revenue companies close

### PR DESCRIPTION
**Pull request by KevinR1000**
Companies P15-HR and P20-C&WR close at beginning of phase 7 if player-owned (in `def event_phase_revenue!`).
Rather than generic statement, report to log the balance and owner of each closing company.
See issue https://github.com/tobymao/18xx/issues/5168 .

### CODE REVIEW

File: lib/engine/game/g_1822/game.rb [hunk 1]
```diff
@@ -727,11 +727,14 @@ def event_full_capitalisation!
...
         def event_phase_revenue!
-          @log << '-- Event: P15-HR and P20-C&WR now closes and its money returned to the bank --'
+          @log << '-- Event: Phase revenue companies now close with money returned to the bank --'
           self.class::PRIVATE_PHASE_REVENUE.each do |company_id|
             company = @companies.find { |c| c.id == company_id }
             next if !company || company&.closed? || !@phase_revenue[company_id]
 
+            @log << "#{company.name} "\
+              "(#{format_currency(@phase_revenue[company_id].cash)}, #{company.owner.name}) "\
+              'closes'
```
Comment (crericha) [inline lib/engine/game/g_1822/game.rb:737]: Why include cash and owner name? I don't believe any other private closing log message includes things like that. my personal preference is for consistency unless good reason.

File: lib/engine/game/g_1822/game.rb [hunk 2]
```diff
@@ -727,11 +727,14 @@ def event_full_capitalisation!
         end
 
         def event_phase_revenue!
-          @log << '-- Event: P15-HR and P20-C&WR now closes and its money returned to the bank --'
+          @log << '-- Event: Phase revenue companies now close with money returned to the bank --'
```
Comment (crericha) [inline lib/engine/game/g_1822/game.rb:730]: "Phase revenue companies" is misleading as P9 also provides phase revenue.

**Comment by KevinR1000** [comment]
"Phase revenue" is the name the Info page uses for the event.
`Phase revenue | P15-HR and P20-C&WR closes if not acquired by a major company`

Probably no good reason to call out the player who made a mistake (abandoning money), but it saves me from having to look back in the history to see who made the mistake.

I have no objection to reverting the current log entry and stripping details out of the new entries.

**Comment by KevinR1000** [comment]
A possibly good reason to use the "phase revenue companies" language:
-  This code is apparently used in 1822 variants where one or both of P15-HR and P20-C&WR do not exist.  See issue.
-  If the language is generic, this code could be used in other 1822 games where similar companies exist.

Neither of those is overwhelming, however.

**Comment by crericha** [comment]
If phase revenue is called out as P15 and P20 on the info page, then its fine.

Yeah, let's not call out the player. We can secretly judge them. ;-)

**Review by crericha — APPROVED** [review]

**Pull request merged**